src/index.js

```diff
@@ -232,8 +247,21 @@ function shimHasOne(target) {
         return original.apply(this, arguments);
       }
 
-      const loader = options[EXPECTED_OPTIONS_KEY].loaders[this.target.name].bySingleAttribute[this.foreignKey];
-      return Promise.resolve(loader.load(instance.get(this.sourceKey)).then(rejectOnEmpty.bind(null, options)));
+      return Promise.resolve().then(() => {
```

Comment:
Refactoring and fixing the paranoid issue for `hasOne` relationship, again using the same idea from `shimModel` function.